shell/platform/android/io/flutter/embedding/android/AndroidTouchProcessor.java

```diff
@@ -354,7 +381,19 @@ private int getPointerChangeForAction(int maskedAction) {
     if (maskedAction == MotionEvent.ACTION_SCROLL) {
       return PointerChange.HOVER;
     }
-    return -1;
+    throw new AssertionError("Unexpected masked action");
+  }
+
+  @PointerChange
+  private int getPointerChangeForPanZoom(int pointerChange) {
+    if (pointerChange == PointerChange.DOWN) {
+      return PointerChange.PAN_ZOOM_START;
+    } else if (pointerChange == PointerChange.MOVE) {
+      return PointerChange.PAN_ZOOM_UPDATE;
+    } else if (pointerChange == PointerChange.UP || pointerChange == PointerChange.CANCEL) {
+      return PointerChange.PAN_ZOOM_END;
+    }
+    throw new AssertionError("Unexpected pointer change");
```

Comment:
Should the errors be caught anywhere, or they will never occur given how Flutter's code is written?

Reply:
This error should never be thrown. I saw some other uses of uncaught `AssertionError` for similar cases.